compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1069,6 +1070,14 @@ class JSCodeGen()(using genCtx: Context) {
       } else if (sym.isJSNativeCtorDefaultParam) {
         // #11592
         None
+      } else if (sym.is(Bridge) && sym.name.is(DefaultGetterName) && currentClassSym.isNonNativeJSClass) {
+        /* #12572 Bridges for default accessors in non-native JS classes must not be emitted,
+         * because they call another default accessor, making their entire body an
+         * <undefined-param> that cannot be eliminated.
```

Comment:
What's `<undefined-param>` ? I see the test cases set the default value to `= js.undefined` but what if I use a different default value?

Reply:
The right-hand-side of a default value is irrelevant. As soon as there is a default value, where `= js.undefined`, `= 42` or `= js.native`, that parameter will be an `<undefined-param>`. `<undefined-param>`'s are markers that are always eliminated at compile-time, and replaced by *no* argument (because you can do that in JS). That is done at
https://github.com/lampepfl/dotty/blob/0d1d47a8fde5226b81376f5ff09512be17abaa0c/compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala#L3764-L3776

Reply:
Ah, so should the compiler disallow a default value that is not exactly `js.undefined` ?

Reply:
No, it shouldn't. It wouldn't work when the type of the parameter is `Int`, for example. Also, an actual value can have valuable documentation purposes. It's been allowed for 8 years in Scala.js for Scala 2, so we can't change that.